### PR TITLE
Tt 29 warn of unsaved changes on customer page

### PR DESCRIPTION
**Pull request by jr-98**
The modal has been implemented and the new edit button flow has been configured.

### CODE REVIEW

File: src/app/modules/customer-management/components/customer-info/components/customer-list/customer-list.component.html [hunk 1]
```diff
@@ -19,20 +14,11 @@
         <td class="col-5 text-break">{{ customer.id }}</td>
         <td class="col-3 text-break">{{ customer.name }}</td>
         <td class="col-4 text-center">
-          <button
-            (click)="editCustomer(customer.id)"
-            type="button"
-            class="btn btn-sm btn-primary"
-          >
+          <button data-toggle="modal" data-target="#editModal" (click)="openEditModal(customer)" type="button" class="btn btn-sm btn-primary">
```
Comment (josepato87) [inline src/app/modules/customer-management/components/customer-info/components/customer-list/customer-list.component.html:17]: Same comment as above regarding big lines of properties.

File: src/app/modules/customer-management/components/customer-info/components/customer-list/customer-list.component.html [hunk 2]
```diff
@@ -19,20 +14,11 @@
...
             <i class="fa fa-pencil fa-xs"></i>
           </button>
-          <button
-            data-toggle="modal"
-            data-target="#deleteModal"
-            (click)="openModal(customer)"
-            type="button"
-            class="btn btn-sm btn-danger ml-2"
-          >
+          <button data-toggle="modal" data-target="#deleteModal" (click)="openModal(customer)" type="button"
```
Comment (josepato87) [inline src/app/modules/customer-management/components/customer-info/components/customer-list/customer-list.component.html:20]: Same as above.

File: src/app/modules/customer-management/components/customer-info/components/customer-list/customer-list.component.html [hunk 3]
```diff
@@ -41,15 +27,9 @@
...
-  aria-hidden="true"
-  [title]="'Delete Customer'"
-  [body]="message"
-  (closeModalEvent)="deleteCustomer()"
->
+<app-dialog *ngIf="showModal" class="modal fade" id="deleteModal" tabindex="-1" role="dialog" aria-hidden="true"
+  [title]="'Delete Customer'" [body]="message" (closeModalEvent)="deleteCustomer()">
+</app-dialog>
+<app-dialog *ngIf="showModal" class="modal fade" id="editModal" tabindex="-1" role="dialog" aria-hidden="true"
+  [title]="'Edit Customer'" [body]="message" (closeModalEvent)="editCustomer()">
```
Comment (josepato87) [inline src/app/modules/customer-management/components/customer-info/components/customer-list/customer-list.component.html:34]: Same as above.

File: src/app/modules/customer-management/components/customer-info/components/customer-list/customer-list.component.ts [hunk 4]
```diff
@@ -109,5 +111,14 @@ export class CustomerListComponent implements OnInit, OnDestroy, AfterViewInit {
     this.message = `Are you sure you want to delete ${item.name}?`;
     this.showModal = true;
   }
-
+  openEditModal(item: Customer) {
+    if (this.showCustomerForm === true) {
+      this.idToEdit = item.id;
```
Comment (josepato87) [inline src/app/modules/customer-management/components/customer-info/components/customer-list/customer-list.component.ts:116]: This line is the same in IF and ELSE clauses, please put it before the IF.

File: src/app/modules/customer-management/components/customer-info/components/customer-list/customer-list.component.html [hunk 5]
```diff
@@ -1,11 +1,6 @@
 <div class="table-responsive-sm">
-  <table
-    *ngIf="customers"
-    class="table table-bordered table-striped mb-0"
-    datatable
-    [dtTrigger]="dtTrigger"
-    [dtOptions]="dtOptions"
-  >
+  <table *ngIf="customers" class="table table-bordered table-striped mb-0" datatable [dtTrigger]="dtTrigger"
```
[the remark this answers is not in the document]
Comment (jr-98) [reply src/app/modules/customer-management/components/customer-info/components/customer-list/customer-list.component.html:2]: Ok. I'm Sorry. What happens is that I have a pluggin 
called beautifull which iedenta automatically and when I save it it has moved the original structure.

**Pull request closed without merging**